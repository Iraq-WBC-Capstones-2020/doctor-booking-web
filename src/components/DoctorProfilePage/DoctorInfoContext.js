import React, { createContext, useState } from 'react';
import Avatar from './ProfileHeader/images/avatar1.svg';

export const InfoContext = createContext();

export function InfoProvider(props) {
  const [DocInfo, setDocInfo] = [
    {
      fullName: 'Salah Mohammed',
      speciality: 'Children doctor',
      experinces:
        "Some quick example text to build on the card title and make up the bulk of the card's content.Some quick example text to build on the card title and make up the bulk of the card's content.",
      educaton:
        "Some quick example text to build on the card title and make up the bulk of the card's content.Some quick example text to build on the card title and make up the bulk of the card's content.",
      photo: Avatar,
      days: [
        { day: 'sunday', date: '11 - 5' },
        { day: 'monday', date: '11 - 3' },
        { day: 'tuseday', date: '12 - 4' },
        { day: 'wedensday', date: '9 - 1' },
        { day: 'thirsday', date: '11 - 5' },
        { day: 'friday', date: 'off day' },
        { day: 'saturday', date: 'off day' },
      ],
    },
  ];

  return (
    <InfoContext.Provider value={[DocInfo, setDocInfo]}>
      {props.children}
    </InfoContext.Provider>
  );
}
