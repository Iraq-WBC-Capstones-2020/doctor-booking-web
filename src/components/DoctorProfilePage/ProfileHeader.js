import React from 'react'
import { Container } from 'react-bootstrap'


function ProfileHeader(){

    return(

        <div className="container-fluid">
            <Container>
                <div className="profileImage">
                    <img src="" alt="" />
                </div>
                <div className="profileName">
                    <h2>
                        Doctor Name
                    </h2>
                </div>
                <div className="bookButton">
                    <Button>

                    </Button>
                </div>
            </Container>
        </div>

    )

}

export default ProfileHeader