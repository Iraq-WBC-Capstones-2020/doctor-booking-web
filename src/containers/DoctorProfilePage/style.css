.theBody {
  margin-top: 5rem;
}

.bodyRow {
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.bookButton {
  width: 100%;
  height: 35px;
  background-color: 026fc2;
}
/*# sourceMappingURL=style.css.map */
