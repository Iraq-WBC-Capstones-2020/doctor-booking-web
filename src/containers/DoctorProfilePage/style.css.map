{
    "version": 3,
    "mappings": "AAAA,AAAA,QAAQ,CAAC;EACP,UAAU,EAAE,IAAI;CACjB;;AAED,AAAA,QAAQ,CAAC;EACP,eAAe,EAAE,UAAU;CAC5B;;AAED,AAAA,WAAW,CAAC;EACV,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,IAAI;EACZ,gBAAgB,EAAE,MAAM;CACzB",
    "sources": [
        "style.scss"
    ],
    "names": [],
    "file": "style.css"
}