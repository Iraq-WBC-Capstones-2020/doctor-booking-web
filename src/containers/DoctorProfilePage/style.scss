.theBody{

    margin-top: 5rem;

}

.bodyRow{
    justify-content: flex-start;
}