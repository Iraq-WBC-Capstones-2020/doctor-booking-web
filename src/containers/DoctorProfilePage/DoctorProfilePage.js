import React, { Fragment } from 'react';
import ProfileHeader from '../../components/DoctorProfilePage/ProfileHeader/ProfileHeader';
import TimeTable from '../../components/DoctorProfilePage/TimeTable/TimeTable';
import Informations from '../../components/DoctorProfilePage/informations/Informations';
import { InfoProvider } from '../../components/DoctorProfilePage/DoctorInfoContext';
import './style.css';
import { Container, Row, Button } from 'react-bootstrap';
import ClinicMap from '../../components/DoctorProfilePage/ClinicMap/ClinicMap';

function DoctorProfilePage() {
  return (
    <InfoProvider>
      <Fragment>
        <ProfileHeader />
        <Container className="theBody">
          <Row className="bodyRow">
            <TimeTable />
            <Informations />
          </Row>
          <ClinicMap />
          <Button className="bookButton mt-5">Book an appointment</Button>
        </Container>
      </Fragment>
    </InfoProvider>
  );
}

export default DoctorProfilePage;
