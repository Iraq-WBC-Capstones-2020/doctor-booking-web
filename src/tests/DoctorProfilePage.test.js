import React from 'react';
import DoctorProfilePage from '../containers/DoctorProfilePage/DoctorProfilePage';
import { renderHook } from '@testing-library/react-hooks';

test('DoctorProfilePage  is rendering  ', () => {
  renderHook(<DoctorProfilePage />);
});

// test('DoctorProfilePage run well ', () => {
//   const tree = renderer.create(<DoctorProfilePage />).toJSON();
//   expect(tree).toMatchSnapshot();
// });
