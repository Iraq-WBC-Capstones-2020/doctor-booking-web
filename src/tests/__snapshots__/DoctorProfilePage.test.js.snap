// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoctorProfilePage run well  1`] = `
Array [
  <div
    className="cover container-fluid d-flex align-items-end"
  >
    <div
      className="profileCover d-flex align-items-center container"
    >
      <div
        className="profileImage"
      >
        <img
          alt="profile"
          src="avatar1.svg"
        />
      </div>
      <div
        className="profileName d-flex flex-column ml-4"
      >
        <h2>
          Salah Mohammed
        </h2>
        <button
          className="btn btn-primary"
          disabled={false}
          type="button"
        >
          Book an appointment
        </button>
      </div>
    </div>
  </div>,
  <div
    className="theBody container"
  >
    <div
      className="bodyRow row"
    >
      <div
        className="timeTable"
      >
        <div
          className="col"
        >
          <div
            className="ml-2 justify-content-start row"
          >
            <h4
              className="specitial"
            >
              Children doctor
            </h4>
          </div>
          <div
            className="ml-2 justify-content-start row"
          >
            <table
              className="table-borderless table"
            >
              <thead>
                <tr>
                  <h4>
                    Time table
                  </h4>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    sunday
                  </td>
                  <td>
                    11 - 5
                  </td>
                </tr>
                <tr>
                  <td>
                    monday
                  </td>
                  <td>
                    11 - 3
                  </td>
                </tr>
                <tr>
                  <td>
                    tuseday
                  </td>
                  <td>
                    12 - 4
                  </td>
                </tr>
                <tr>
                  <td>
                    wedensday
                  </td>
                  <td>
                    9 - 1
                  </td>
                </tr>
                <tr>
                  <td>
                    thirsday
                  </td>
                  <td>
                    11 - 5
                  </td>
                </tr>
                <tr>
                  <td>
                    friday
                  </td>
                  <td>
                    off day
                  </td>
                </tr>
                <tr>
                  <td>
                    saturday
                  </td>
                  <td>
                    off day
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <div>
        <div
          className="row"
        >
          <div
            className="col-lg-8"
          >
            <div
              className="informations"
            >
              <div
                className="informationsCards education"
              >
                <div
                  className="infCards card border-light"
                  style={
                    Object {
                      "width": "45rem",
                    }
                  }
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="card-title h5"
                    >
                      Education
                    </div>
                    <p
                      className="cardText card-text"
                    >
                      Some quick example text to build on the card title and make up the bulk of the card's content.Some quick example text to build on the card title and make up the bulk of the card's content.
                    </p>
                  </div>
                </div>
              </div>
              <div
                className="informationsCards experinces"
              >
                <div
                  className="infCards card border-light"
                  style={
                    Object {
                      "width": "45rem",
                    }
                  }
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="card-title h5"
                    >
                      Experinces
                    </div>
                    <p
                      className="cardText card-text"
                    >
                      Some quick example text to build on the card title and make up the bulk of the card's content.Some quick example text to build on the card title and make up the bulk of the card's content.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mt-5 theAdress"
    >
      <h3
        className="mt-5 mb-5"
      >
        Adress
      </h3>
      <h6>
        city, country
      </h6>
      <div
        style={
          Object {
            "cursor": "grab",
            "height": 0,
            "position": "relative",
            "width": 0,
          }
        }
      >
        <div
          style={
            Object {
              "height": "100%",
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            className=""
            style={
              Object {
                "height": "100%",
                "overflow": "hidden",
                "position": "absolute",
                "visibility": "inherit",
                "width": "100%",
              }
            }
          />
          <div
            style={
              Object {
                "height": 0,
                "overflow": "visible",
                "width": 0,
              }
            }
          />
        </div>
      </div>
    </div>
    <button
      className="bookButton mt-5 btn btn-primary"
      disabled={false}
      type="button"
    >
      Book an appointment
    </button>
  </div>,
]
`;
